src/app/donation-start/donation-start-form/donation-start-form.component.html

```diff
@@ -621,7 +621,7 @@ <h3 class="b-rt-0 b-m-0 b-bold"><span class="span-hr"></span>How does Big Give u
                 </tr>
               </table>
             }
-            @if (!submitting) {
+            @if (!submitting && !runningFinalPreSubmitUpdate) {
```

Comment:
Haven't read all the way through but if I'm understanding right we need to prevent the donation being made until the final pre submit update is complete. So I think it's more robust to make this like 

```suggestion
            @if (!submitting && finalPreSubmitComplete) {
```

`finalPreSubmitComplete` would be a property that is initialized to false and set true when we get a successful return from matchbot.

Reply:
Not sure – it seems harder to track robustly because donors could go back to earlier steps and we want to do this every time they [re-]enter Confirm.

Reply:
I don't think it's a critical issue.